api/infrastructure/v1alpha1/gardenershootcluster_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GSCReferenceNamespaceKey     = "infrastructure.cluster.x-k8s.io/gsc_namespace"
+	GSCReferenceNameKey          = "infrastructure.cluster.x-k8s.io/gsc_name"
+	GSCReferecenceClusterNameKey = "controlplane.cluster.x-k8s.io/gsc_cluster"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// GardenerShootClusterSpec defines the desired state of GardenerShootCluster.
+type GardenerShootClusterSpec struct {
+	// CloudProfile contains a reference to a CloudProfile or a NamespacedCloudProfile.
+	// +optional
+	CloudProfile *v1beta1.CloudProfileReference `json:"cloudProfile,omitempty"`
+}
+
+// GardenerShootClusterStatus defines the observed state of GardenerShootCluster.
+type GardenerShootClusterStatus struct {
+	// Ready denotes that the foo cluster infrastructure is fully provisioned.
```

Comment:
Is the `foo` from the generation?

```suggestion
	// Ready denotes that the GardenerShootCluster is fully provisioned.
```

Reply:
Whoopsie 🤭
This is a copy-paste from the CAPI-book.

Reply:
In [497289f](https://github.com/LucaBernstein/cluster-api-provider-gardener/pull/13/commits/497289f5de3ba1a94e1d6b955e63470b3141f775), I adapted the status update of the `GardenerShootCluster` in a way that it syncs to the readiness of the Seed where the resulting shoot will be hosted.
Status is currently only posted when the Shoot was scheduled to a seed. Maybe we can improve this in the future.